mautrix-whatsapp: 0.1.7 -> 0.1.8

chvp opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Package bump.

###### Things done

I've deployed this to my matrix server, everything still works as expected.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

chvp committed:
mautrix-whatsapp: 0.1.7 -> 0.1.8

r-rmcgibbo commented:
Result of `nixpkgs-review pr 133015` at bb550a67 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>mautrix-whatsapp</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`agpl3` is a deprecated license, please check if project uses `agpl3Plus` or `agpl3Only` and change `meta.license` accordingly.

Near pkgs/servers/mautrix-whatsapp/default.nix:25:5:
```
   |
25 |     license = licenses.agpl3;
   |     ^
```
</li>
  </ul>
</details>

chvp committed:
mautrix-whatsapp: update repository location

chvp committed:
mautrix-whatsapp: specify agpl3plus license

e.g. https://github.com/mautrix/whatsapp/blob/f4a7394b670db59aadd6ad99f761acbff255e1b6/bridgestate.go
mentions the option of using a later version.

Ma27 reviewed (commented):
Diff LGTM. Cannot deploy this to my homeserver since Hetzner is having issues :) 
As soon as it's possible again and the change turns out to be fine, I'll merge.
